rad/src/operators/string.rs

```diff
@@ -35,6 +38,117 @@ pub fn parse_json_array(input: &RadonString) -> Result<RadonArray, RadError> {
+                // <tag>Hello</tag>
+                rad_array.value().into_iter().next().unwrap()
+            } else {
+                // <tag>Hello\nWorld</tag>
+                rad_array.clone().into()
+            }
+        }
+        _ => {
+            let mut new_map: BTreeMap<String, RadonTypes> = BTreeMap::new();
+            for k in input.keys() {
+                match input.get(k) {
+                    Some(RadonTypes::Array(rad_array)) if rad_array.value().len() == 1 => {
+                        let value = rad_array.value().into_iter().next().unwrap();
+                        new_map.insert(k.clone(), value);
+                    }
+                    Some(value) => {
+                        new_map.insert(k.clone(), value.clone());
+                    }
+                    None => unreachable!("Invalid map"),
+                }
+            }
+
+            RadonMap::from(new_map).into()
+        }
+    }
+}
+
+pub fn parse_element_map(
+    input: &minidom::Element,
+    use_namespace: bool,
+    depth: u8,
+) -> Result<RadonTypes, RadError> {
+    if depth > MAX_DEPTH {
+        return Err(RadError::XmlParseOverflow);
+    }
+
+    let mut map: BTreeMap<String, RadonTypes> = BTreeMap::new();
+    if use_namespace {
+        map.insert("_xmlns".to_string(), RadonString::from(input.ns()).into());
+    }
+    for (k, v) in input.attrs() {
+        map.insert(format!("@{}", k), RadonString::from(v).into());
+    }
+
+    for child in input.nodes() {
+        let (ignore, key, value) = match child {
+            minidom::Node::Element(elem) => {
+                let key = elem.name().to_string();
+                let value = parse_element_map(elem, false, depth + 1)?;
+
+                (false, key, value)
+            }
+            minidom::Node::Text(text) => {
+                // This check is to avoid blank spaces in xml would be included in the RadonMap
+                let text_var = text.trim().to_string();
+                if !text_var.is_empty() {
+                    (
+                        false,
+                        "_text".to_string(),
+                        RadonString::from(text_var).into(),
+                    )
+                } else {
+                    (true, "".to_string(), RadonString::from("").into())
+                }
+            }
+        };
+
+        match (ignore, map.remove(&key)) {
+            (true, _) => continue,
+            (false, None) => {
+                map.insert(key, RadonArray::from(vec![value]).into());
+            }
+            (false, Some(RadonTypes::Array(rad_array))) => {
+                let mut new_array = rad_array.value();
+                new_array.push(value);
+                map.insert(key, RadonArray::from(new_array).into());
+            }
+            _ => unreachable!("Invalid XML parsing"),
+        }
+    }
+
+    Ok(magic_map_cleaning(map))
+}
+
+pub fn parse_xml_map(input: &RadonString) -> Result<RadonMap, RadError> {
```

Comment:
Maybe it's worth to add a link to the WIP, because it explains the parsing process?